### ISSUE TITLE
ol.style.Icon disappears when setStyle is called

### ISSUE THREAD

**Issue by MadalenaGoncalves** [openlayers/openlayers#14193]
I'm updating the styles of a feature rendered with ol.Icon, as a response to a click event. With each click, a new svg is generated and the feature styles are updated.
On Firefox, the feature styles are "lost" in the first clicks, but after a few more clicks it works as it should.

**Example**: https://jsfiddle.net/madalenagoncalves/f4sqnkgt/4/

**Current behavior**: On Firefox, the feature styles are not correctly applied when generating new icons with ol.style.Icon.

**Expected behavior**: The feature styles should always be correctly applied.

The example breaks on Firefox but not on Chrome. However, in my original code (with a more complex svg), the issue happens also on Chrome.

This only seems to happen with ol.style.Icon, as the ol.style.Circle in the example updates correctly.

_Originally posted by @MadalenaGoncalves in https://github.com/openlayers/openlayers/issues/5412#issuecomment-1274479120_

**Comment by mike-000** [comment]
That is using a very old version of OpenLayers.  It works correctly if you use the latest version https://jsfiddle.net/02zyhjvo/

**Comment by MadalenaGoncalves** [comment]
Thanks for the reply. Your suggestion worked :)

I had tried passing src to the ol.style.Icon before, but then I got a typescript error I could not pass a string when an Image was expected.
I realise now see that the error was not triggered by ol.style.Icon. I was missing the return type on the method that created the image, and when I changed from returning an image to returning just the src string, typescript did not recognise that the return type had changed. By explicitly setting the return type in the method that generates the src, the error was gone and everything worked.

This approach now causes the icons to flicker, but I guess that's expected and can be solved by caching the icons.
